### ISSUE TITLE
[Regression] Icon color inspector in singlestat broken

### ISSUE THREAD

**Issue by Krinkle** [grafana/grafana#2953]
In upgrading from v1.9 to v2.1.3 the icon color inspector regressed. It's no longer possible to update values using the user interface. The preview color is correct, but is never acknowledged in the data model.

Clicking invert or refreshing the dashboard restores the original value.

I can reproduce this on latest master as well when using http://play.grafana.org/dashboard/db/new-features-in-v19?panelId=18&fullscreen&edit

1. Click on a color (e.g. the green one).
2. Change it.
3. Click elsewhere to close the color inspector.
4. Refresh the graph or click "Invert".

Actual result:

The original color (green) is still used.

**Comment by woodsaj** [comment]
I am unable to replicate this in chrome on Linux. @Krinkle what OS, browser are you using?

**Comment by torkelo** [comment]
yea, I also failed to replicate in chrome and in firefox. Can you check if you get any js errors in the chrome dev console?

**Comment by Krinkle** [comment]
I can still consistently reproduce this at <http://play.grafana.org/dashboard/db/new-features-in-v19?panelId=18&fullscreen&edit> in Chrome 46.0.2490.71 (stable) as well as Chrome 48.0.2538.0 (canary), Mac OS X 10.10.5. No errors or warnings in the console of any kind.

> 1. Click on a color (e.g. the green one).
<img width="243" alt="screen shot" src="https://cloud.githubusercontent.com/assets/156867/10562231/83e8821c-7548-11e5-94e9-39556185dac1.png">
> 2. Change it.
<img width="270" alt="screen shot" src="https://cloud.githubusercontent.com/assets/156867/10562229/83c441cc-7548-11e5-91e0-83914319fab2.png">
> 3. Click elsewhere to close the color inspector.
<img width="276" alt="screen shot" src="https://cloud.githubusercontent.com/assets/156867/10562230/83dd389e-7548-11e5-8746-59442a4c9e2f.png">
> 4. Refresh the graph or click "Invert".
<img width="255" alt="screen shot" src="https://cloud.githubusercontent.com/assets/156867/10562232/856dd48e-7548-11e5-8ae8-5b2b6885627b.png">


Result:

The original color (green) is still used. The update was never saved in the model.

**Comment by woodsaj** [comment]
@krinkle demonstrated this bug to me at Velocity.  I can confirm he is not making it up :)

**Comment by torkelo** [comment]
strange, works in chrome 42 and firefox 37

**Comment by torkelo** [comment]
tested on chrome 46 on mac os x on play.grafana.org and could not replicate it, not sure how you do it :(

**Comment by torkelo** [comment]
any idea for how to help me replicate it? Can you replicate it in a colleagues browsers? Maybe some browser extension that is causing it?

**Comment by Krinkle** [comment]
I've reproduced it on my Mac with Chrome 48 canary, Chrome 46 stable, and plain Chromium 48 dev. I can also reproduce it with Chrome 46 stable on Windows 10 in BrowserStack.

Below is a recording from that BrowserStack session:
![Screen recording](https://cloud.githubusercontent.com/assets/156867/10645503/ef16e9c2-7825-11e5-99ef-2702a1b33d15.gif)

**Comment by Krinkle** [comment]
Were you able to reproduce this bug? It's quite the nuisance and a source of confusion from users to not be able to change colours of singlestats.

Especially following the Grafana 2.x upgrade where themes (light, dark) are now user controlled. As such, we find ourselves having to change lots of dashboards where bright or dark colours were used for text (which is unreadable in the wrong theme). Having a set of predefined colours to choose from (like for metrics) would be useful here. It would shade the selected colour light or dark depending on the user's theme.

**Comment by torkelo** [comment]
@Krinkle I am sorry, so far I have not found a way to replicate the issue on any computer / os.

**Comment by Krinkle** [comment]
Several people have reported this and all able to reproduce it (Windows/Mac, Chrome/Firefox). I can only ask you to look closely at the capture. Alternatively, show me how you are able to successfully change the color of a singlestat in the user interface. Does the preview use the new color?

**Comment by torkelo** [comment]
@Krinkle several? I am only aware of this issue, and only your report. Its not that I don't believe you! 

But have tried all OSs and browsers and it works fine for me so really hard to fix it. 

Gif animation (colors are scrambled because of gif compression) 
![singlestat](https://cloud.githubusercontent.com/assets/10999/10932542/f13d83e4-82d1-11e5-9a4f-50ca2cea0f4c.gif)

**Comment by Krinkle** [comment]
@torkelo Several people at Wikimedia reported it to me. However after persisting to have more people try it, I eventually found two people that couldn't reproduce it.

The key is in how the color inspector is closed.

If one closes it by clicking on the empty space around the inspector everything works fine.

However, if you:
1. Close the inspector by clicking on the same color square that opened it.
2. Close the inspector by clicking on another color square and opening that one.

.. then the bug happens. That little extra click outside the inspector is a step you did, but I did not.
